File: kernel/src/test/java/com/sos/scheduler/engine/kernel/test/EnvironmentFilesTest.java
package com.sos.scheduler.engine.kernel.test;

import static com.google.common.io.Files.createTempDir;
import static com.sos.scheduler.engine.kernel.util.Classes.springPattern;
import static com.sos.scheduler.engine.kernel.util.Files.removeFile;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.arrayContainingInAnyOrder;
import static org.hamcrest.Matchers.arrayWithSize;
import static org.hamcrest.Matchers.emptyArray;

import java.io.File;
import java.io.IOException;

import org.junit.Test;
import org.springframework.core.io.Resource;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;

import com.sos.scheduler.engine.kernel.util.ResourcePath;

public final class EnvironmentFilesTest {
    private static final String[] expectedNames = {"scheduler.xml", "factory.ini", "sos.ini"};

    @Test public void test() {
        File dir = createTempDir();
        try {
            assertThat(dir.list(), emptyArray());
            EnvironmentFiles.copy(new ResourcePath(EnvironmentFilesTest.class.getPackage(), "config"), dir);
            assertThat(dir.list(), arrayContainingInAnyOrder(expectedNames));
        } finally {
            for (String name: expectedNames) removeFile(new File(dir, name));
            removeFile(dir);
        }
    }

    @Test public void testSpringGetResources() throws IOException {
        PathMatchingResourcePatternResolver r = new PathMatchingResourcePatternResolver();
        Resource[] result = r.getResources(springPattern(EnvironmentFilesTest.class.getPackage(), "config/scheduler.xml"));
        assertThat(result, arrayWithSize(1));
    }
}

File: opensource/test/src/test/scala/com/sos/scheduler/engine/test/EnvironmentFilesTest.scala
package com.sos.scheduler.engine.test

import EnvironmentFilesTest._
import com.google.common.io.Files.createTempDir
import com.sos.scheduler.engine.common.system.Files.removeFile
import com.sos.scheduler.engine.kernel.util.Classes.springPattern
import com.sos.scheduler.engine.kernel.util.ResourcePath
import java.io.File
import org.junit.runner.RunWith
import org.scalatest.FunSuite
import org.scalatest.junit.JUnitRunner
import org.scalatest.matchers.ShouldMatchers._
import org.springframework.core.io.support.PathMatchingResourcePatternResolver

@RunWith(classOf[JUnitRunner])
final class EnvironmentFilesTest extends FunSuite {

  test("copy") {
    val dir = createTempDir()
    try {
      dir.list.toSet should be ('empty)
      TestEnvironmentFiles.copy(new ResourcePath(classOf[EnvironmentFilesTest].getPackage, "config"), dir)
      dir.list.toSet should equal (expectedNames)
    }
    finally {
      for (name <- expectedNames) removeFile(new File(dir, name))
      removeFile(dir)
    }
  }

  test("Spring getResources") {
    val r = new PathMatchingResourcePatternResolver
    val result = r.getResources(springPattern(classOf[EnvironmentFilesTest].getPackage, "config/scheduler.xml"))
    result should have size(1)
  }
}

private object EnvironmentFilesTest {
  val expectedNames = Set("scheduler.xml", "factory.ini", "sos.ini")
}
